master-firmware/src/base/base_controller.c

```diff
@@ -264,3 +266,58 @@ void trajectory_manager_start(void)
     static THD_WORKING_AREA(trajectory_thd_wa, TRAJECTORY_MANAGER_STACKSIZE);
     chThdCreateStatic(trajectory_thd_wa, sizeof(trajectory_thd_wa), TRAJECTORY_MANAGER_PRIO, trajectory_manager_thd, &(robot.traj));
 }
+
+static struct {
+    int x_mm;
+    int y_mm;
+    int a_deg;
+} goal;
+
+void motion_planner_set_goal(int x_mm, int y_mm)
+{
+    goal.x_mm = x_mm;
+    goal.y_mm = y_mm;
+}
+
+static THD_FUNCTION(motion_planner_thd, arg)
+{
+    (void)arg;
+    chRegSetThreadName(__FUNCTION__);
+
+    while (1) {
+        chThdSleepMilliseconds(100);
+
+        const point_t start = {position_get_x_float(&robot.pos), position_get_y_float(&robot.pos)};
+        if (((start.x - goal.x_mm) * (start.x - goal.x_mm) + (start.y - goal.y_mm) * (start.y - goal.y_mm)) < 100) {
+            NOTICE("Close enough!");
```

Comment:
This will spam the logs with NOTICE at 10 Hz, thats not really great from a debug point of view.

Reply:
:+1: